src/components/DropDownSortContainer.ts

```diff
@@ -70,45 +72,47 @@ export default class DropDownSortContainer extends Component<WrapperProps, DropD
             let targetListView: ListView | null = null;
 
             if (targetNode) {
-                this.setState({ targetNode });
                 targetListView = dijitRegistry.byNode(targetNode);
                 if (targetListView) {
-                    this.setState({ targetListView });
+                    if (!targetListView.__customWidgetDataSourceHelper) {
+                        try {
+                            targetListView.__customWidgetDataSourceHelper = new DataSourceHelper(targetListView);
+                        } catch (error) {
+                            this.setState({
+                                alertMessage: error.message,
+                                targetListView,
+                                targetNode
+                            });
+                        }
+                    } else if (!DataSourceHelper.checkVersionCompatible(targetListView.__customWidgetDataSourceHelper.version)) {
+                        this.setState({
+                            alertMessage: "DataSource compatibility issue"
+                        });
+                    }
+                    this.dataSourceHelper = targetListView.__customWidgetDataSourceHelper as DataSourceHelper;
```

Comment:
Is necessary since `__customWidgetDataSourceHelper` is optional on the targetListView. Unfortunately typescript is not detecting that by line 92 `__customWidgetDataSourceHelper` will already be set a value. When I remove the casting, TS complains that.
`Type 'DataSourceHelper | undefined' is not assignable to type 'DataSourceHelper'.
  Type 'undefined' is not assignable to type 'DataSourceHelper'.`